### PR TITLE
src: read ndarrays in C-contiguous order

### LINKED ISSUE #395: Issue with transposed NumPy arrays
As NumPy does transposition by changing the stride, the following calls gives the same resulting APyTypes array:

```
import numpy as np
from apytypes import APyFixedArray
a = np.asarray([[1, -1], [1, -1]])
a1 = APyFixedArray(a, 2, 3)
a2 = APyFixedArray(a.T, 2, 3)
assert a1.is_identical(a2)  # Should not be true
```

Solution is either to use strides in `APyTypes` (which is useful for fast transposition, but probably also other things...) and copy the stride of the input array, or take the stride into consideration when reading data from the NumPy array.

### PR DESCRIPTION
**Pull request by miklhh**
Closes #395

### CODE REVIEW

**Comment by oscargus** [comment]
I think the test should be to use `a.T`, right? Not transposing the apytypes array?

**Comment by oscargus** [comment]
And it doesn't close the issue since this is still a problem for `APyFloatArray`.

**Comment by miklhh** [comment]
> I think the test should be to use `a.T`, right? Not transposing the apytypes array?

Good catch, thank you!

File: lib/apytypes/_apytypes.pyi [hunk 1]
```diff
@@ -80,8 +77,8 @@ class APyFixed:
         """
         Create an :class:`APyFixed` object from :class:`str`.
 
-        Exactly two of three bit-specifiers (*bits*, *int_bits*, *frac_bits*) needs
-        to be set.
+        Exactly two of three bit-specifiers (`bits`, `int_bits`, `frac_bits`) must be
```
Comment (oscargus) [inline lib/apytypes/_apytypes.pyi:80]: `*` is the correct way.

**Comment by oscargus** [comment]
I stand corrected. The docs says https://numpydoc.readthedocs.io/en/latest/format.html#parameters so it should be single backtick. (In Matplotlib we have redefined single backtick to automatically interpret it as `:class:`, `:func:`, etc.)

**Pull request merged**